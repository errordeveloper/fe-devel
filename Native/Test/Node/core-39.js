/*
 *  Copyright 2010-2012 Fabric Engine Inc. All rights reserved.
 */

FABRIC = require('Fabric').createClient();

var Vec2 = function( x, y ) {
  if ( typeof x === "number" && typeof y === "number" ) {
    this.x = x;
    this.y = y;
  }
  else if ( x === undefined && y === undefined ) {
    this.x = 0;
    this.y = 0;
  }
  else throw "new Vec2: invalid arguments";
};

Vec2.prototype = {
  sum: function() {
    return this.x + this.y;
  }
};

var desc = {
  members: { x:'Scalar', y:'Scalar' },
  constructor: Vec2,
  klBindings: {
    filename: "(inline)",
    sourceCode: "\
function Vec2(Scalar x, Scalar y)\n\
{\n\
  this.x = x;\n\
  this.y = y;\n\
}\n\
"
  }
};

FABRIC.RegisteredTypesManager.registerType( 'Vec2', desc );

var node = FABRIC.DG.createNode("foo");
node.addMember("bar", "Vec2");
node.setData("bar", {x:3.5});

FABRIC.flush();
FABRIC.close();
