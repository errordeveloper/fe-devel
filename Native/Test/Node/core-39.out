
node.js:201
        throw e; // process.nextTick error, or 'error' event on first tick
              ^
Fabric core exception: DG.foo.setData('{"memberName":"bar","sliceIndex"...'): missing member 'y'
